instrumentation/net/http/otelhttp/stats.go

```diff
@@ -0,0 +1,206 @@
+var (
+	// Method is the HTTP method of the request, capitalized (GET, POST, etc.).
+	Method = label.Key("http.method")
+	// Host is the value of the HTTP Host header.
+	//
+	// The value of this tag can be controlled by the HTTP client, so you need
+	// to watch out for potentially generating high-cardinality labels in your
+	// metrics backend if you use this tag in views.
+	Host = label.Key("http.host")
+
+	// Scheme is the URI scheme identifying the used protocol: "http" or "https"
+	Scheme = label.Key("http.scheme")
+
+	// StatusCode is the numeric HTTP response status code,
+	// or "error" if a transport error occurred and no status code was read.
+	StatusCode = label.Key("http.status")
+
+	// Path is the URL path (not including query string) in the request.
+	//
+	// The value of this tag can be controlled by the HTTP client, so you need
+	// to watch out for potentially generating high-cardinality labels in your
+	// metrics backend if you use this tag in views.
+	Path = label.Key("http.path")
+)
+
+// Client tag keys.
+var (
+	// KeyClientMethod is the HTTP method, capitalized (i.e. GET, POST, PUT, DELETE, etc.).
+	KeyClientMethod = label.Key("http_client_method")
+	// KeyClientPath is the URL path (not including query string).
+	KeyClientPath = label.Key("http_client_path")
+	// KeyClientStatus is the HTTP status code as an integer (e.g. 200, 404, 500.), or "error" if no response status line was received.
+	KeyClientStatus = label.Key("http_client_status")
+	// KeyClientHost is the value of the request Host header.
+	KeyClientHost = label.Key("http_client_host")
+	// KeyClientScheme is the URI scheme identifying the used protocol: "http" or "https"
+	KeyClientScheme = label.Key("http_client_host")
+)
+
+func (trans *statTransport) applyConfig(c *config) {
+	trans.base.applyConfig(c)
+
+	trans.meter = c.Meter
+	trans.createMeasures()
+}
+
+// RoundTrip implements http.RoundTripper, delegating to Base and recording stats for the request.
+func (trans *statTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	labels := label.NewSet(
+		// conform open-telemetry label definition
+		Method.String(req.Method),
+		Host.String(req.Host),
+		Scheme.String(req.URL.Scheme),
+		Path.String(req.URL.Path),
+		// for prometheus
+		KeyClientMethod.String(req.Method),
+		KeyClientHost.String(req.Host),
+		KeyClientScheme.String(req.URL.Scheme),
+		KeyClientPath.String(req.URL.Path),
+	)
+
+	ctx := req.Context()
+	track := &tracker{
+		start:          time.Now(),
+		ctx:            ctx,
+		valueRecorders: trans.valueRecorders,
+		labels:         &labels,
+	}
+
+	// Perform request.
+	resp, err := trans.base.RoundTrip(req)
+	if err != nil {
+		track.statusCode = http.StatusInternalServerError
+		track.end()
+	} else {
+		track.statusCode = resp.StatusCode
+		if resp.Body == nil {
+			track.end()
+		} else {
+			track.body = resp.Body
+			resp.Body = wrappedBodyIO(track, resp.Body)
```

Comment:
Reading of the response body should not be included in the measurement of a requests duration.

Reply:
The request will be counted as ended after we finish reading all the data from the response body, isn't it?

Reply:
There is probably utility in measurements of both time to first byte and time to the end of the body being read, though the latter is potentially less a measure of the client and more of whatever is processing the response data.

Reply:
This only wrap and returns reader interface rather than actual read and parsing actual data itself. The actual parsing time could be useful, but should not be included in this package, dont you think ?